$OpenBSD: patch-src_sighandler_c,v 1.5 2017/09/13 12:50:33 dcoppa Exp $

commit 09ee12d8e5e1cd219493fde3393dbca10c0cc23f
Author: hwangcc23 <hwangcc@csie.nctu.edu.tw>
Date:   Thu Aug 31 14:48:33 2017

Properly initialize sigaction struct

The code in handle_signal() wasn't clearing the struct sigaction
before passing it to sigaction().
This meant that we would block a random set of signals while executing
the default handler, or jump to the uninitialized __sa_sigaction__
(instead of sa_handler).
Initialize properly as we do in setup_signal_handler().

Index: src/sighandler.c
--- src/sighandler.c.orig
+++ src/sighandler.c
@@ -305,6 +305,8 @@ void handle_signal(int sig, siginfo_t *info, void *dat
 
     struct sigaction action;
     action.sa_handler = SIG_DFL;
+    action.sa_flags = 0;
+    sigemptyset(&action.sa_mask);
     sigaction(sig, &action, NULL);
     raised_signal = sig;
 
